docs: add terminology document for key Veraison concepts (#67)
veraison/docs: terminology.md

# Veraison Terminology

This document explains the key concepts and terms used in **Project Veraison**.  
It is intended to help new contributors, developers, and users quickly understand the terminology.

---

## 🔹 Provisioning

### **CoRIM (Concise Reference Integrity Manifest)**

> A data structure that provides trusted reference values (baselines) for attestation.  

Reviewer: ```suggestion
A data format that provides trusted Reference Values (baselines) and Endorsements for remote attestation.  
```

> It defines what "good" looks like for a platform or component.  

Reviewer: ```suggestion
It defines what are "golden values" for a platform or component.  
```

*(Defined in [IETF CoRIM draft](https://datatracker.ietf.org/doc/draft-birkholz-rats-corim/))*  

### **CoMID (Concise Module Identifier)**
A component of CoRIM that describes the identity and attributes of specific hardware or software modules.  
It enables precise matching of evidence with reference values.  

### **Endorsements**
Statements issued by manufacturers or trusted authorities about a device/component, describing capabilities, identity, or measurement expectations.

---

## 🔹 Verification

### **Evidence**
Claims collected from a device (the Attester) about its current state (firmware, boot configuration, runtime properties).  

### **Scheme**

> A verification policy or method supported by Veraison that defines how evidence should be interpreted and matched against reference values.  

Reviewer: ```suggestion
A verification policy or method supported by Veraison that defines how Evidence should be interpreted and matched against Reference Values.  
```

### **Verifier**

> The entity that validates evidence against endorsements and reference values to produce an attestation result.  

Reviewer: ```suggestion
The entity that validates Evidence against Endorsements and Reference Values to produce an Attestation Result.  
```

### **Attestation Result**
The outcome of the verification process, usually expressing whether the evidence is trustworthy.  

---

## 🔹 Attestation Roles (from IETF RATS)

### **Attester**

> The device or system that generates evidence about its state.  

Reviewer: ```suggestion
The device or system that generates Evidence about its state.  
```

### **Verifier**

> The system (in Veraison) that checks evidence and produces results.  

Reviewer: ```suggestion
An entity (typically a collection of components in Veraison) that  appraise Evidence and produces Attestation Results.  
```

### **Relying Party**

> The consumer of the attestation result (e.g., a cloud service that decides access based on trust).  

Reviewer: ```suggestion
The consumer of the Attestation Result (e.g., a cloud service that decides access based on trust).  
```
